docs/source/howto/codes.rst

```diff
@@ -45,8 +45,90 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>` that knows how to transform the input nodes into the input files that the code expects, copy everything in the code's machine, run the calculation and retrieve the results.
+This plugin would have already been created, either by you (see section on :ref:`interfacing external codes <how-to:codes:plugin>`) or by other AiiDA plugin-developers (available packages are listed in the `plugin registry <https://aiidateam.github.io/aiida-registry/>`_), and has been properly installed in your environment.
+You can check which ones you have currently available:
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+As an example, we will show how to use the ``arithmetic.add`` plugin, which is a pre-installed plugin implemented in `bash` to sum two integers.
+You can access it with the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+Next, we provide the inputs for the code when running the calculation.
+Use ``verdi plugin`` to determine what inputs a specific plugin expects:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
+    (...)
+        Inputs:
+               code:  required  Code        The `Code` to use for this job.
+                  x:  required  Int, Float  The left operand.
+                  y:  required  Int, Float  The right operand.
+    (...)
+
+You will see that 3 inputs nodes are required: two containing the values to add up (``x``, ``y``) and one containing information about the specific code to execute (``code``).
+If you already have these nodes in your database, you can get them by :ref:`querying for them <how-to:data:finding-data>` or using ``orm.load_node(<PK>)``.
+Otherwise, you will need to create them as shown below (note that you `will` need to already have the ``localhost`` computer configured):
+
+.. code-block:: python
+
+    from aiida import orm
+    code_node = orm.Code(remote_computer_exec=[localhost, '/bin/bash'])
+    numx_node = orm.Int(17)
+    numy_node = orm.Int(11)
```

Comment:
In other sections we have not used the suffix `_node` so far. I'd prefer to drop that everywhere.